mancala/protocol/src/lib.rs

```diff
@@ -0,0 +1,131 @@
+//The idea behind these objects is that we should be able to represent a typical mancala game in a
+//concise and easily serializiable manner.  These objects should be ale to be passed through a TCP
+//socket opened between the client and the server without much overhead.
+//
+//The game state can be easily represented by modeling the "board", which consists of two sides and
+//two goals.  The two sides will have several "SLOTS", which in turn will be occuped by zero or more
+//"stones".  The goals each belong to a certain player and are only reachable by that player.  The
+//game moves forwards by a player deciding to move the stones in one of their SLOTS one at a time
+//around the board, depositing them in the following SLOTS one at a time in a counter-clockwise
+//motion.
+//
+//Any time a stone is passed over a player's goal (the slot at the right hand side of either player)
+//that stone is deposited there and a point is scored.  Any time a turn ends in a point scored, the
+//player is allowed to play again.  Players also score points if they end their turn in an otherwise
+//empty slot ON THEIR SIDE that lies directly across from a non-empty opponent slot.  In that case, the
+//last stone placed as well as all of the stones from the opposing slot are scored for the player who
+//made the move.
+
+//Possible workflow for moves:
+//
+// - Poll for "is it my turn"
+// - Once true, Player sends a slot they'd like to move
+// - Slot is checked against possible moves and accepted or rejected
+//    - checked by client or server?
+//        - basic bounds checked by client (does it exist on my side of the board)
+//        - probably server for any check dependent on game state (was the slot i picked empty)
+
+// remove magic numbers
+const SLOTS: usize = 7; // there are 6 playable slots and one goal slot
+const STARTING_STONES: u8 = 4;
+
+#[derive(Debug, Clone)]
+struct GameState {
+    player_one: String,
+    player_two: String,
+    game_board: [u8; SLOTS * 2],
+    player_one_goal_slot: usize,
+    player_two_goal_slot: usize,
+    player_one_turn: bool,
+}
+
+impl GameState {
+    fn new(p_one: String, p_two: String) -> GameState {
+        let mut init_game_board = [STARTING_STONES; SLOTS * 2];
+        init_game_board[SLOTS + 1] = 0;
+        init_game_board[0] = 0;
```

Comment:
are these initializing the goal slots? if so, line 45 should be `init_game_board[SLOTS] = 0;`

Reply:
yup good catch